### ISSUE TITLE
.NET 7 requirement?

### ISSUE THREAD

**Issue by ianleeder** [patrickdemooij9/SeoToolkit.Umbraco#143]
My project is targeting .NET 6 (since it's LTS, and I'm using Umbraco 10 LTS), and I currently have SEO Toolkit 2.2.0 installed.

I'm using both `SeoToolkit.Umbraco` and `SeoToolkit.Umbraco.Core` in different projects, and I see that both packages v3.0.0 require `net7.0`.  Firstly, was this upgrade actually required?  The readme says:
> SeoToolkit is a SEO package for Umbraco 9 & 10 & 11

But users running older versions of Umbraco are most likely running older versions of .NET Core.

Secondly, some weirdness.  

I can't upgrade `SeoToolkit.Umbraco.Core` to 3.0.0 because it says it has a dependency on `net7.0`:
```
NU1202: Package SeoToolkit.Umbraco.Core 3.0.0 is not compatible with net6.0 (.NETCoreApp,Version=v6.0). Package SeoToolkit.Umbraco.Core 3.0.0 supports: net7.0 (.NETCoreApp,Version=v7.0)
Package restore failed. Rolling back package changes for 'Core'.
```
Seems fair enough.

However I **can** upgrade `SeoToolkit.Umbraco` to 3.0.0, even though it says it has a dependency on `net7.0`.
```
Installing NuGet package SeoToolkit.Umbraco 3.0.0.
Generating MSBuild file C:\Users\IanLeeder\source\repos\Website\Umbraco\obj\Umbraco.csproj.nuget.g.props.
Writing assets file to disk. Path: C:\Users\IanLeeder\source\Website\Umbraco\obj\project.assets.json
Successfully uninstalled 'SeoToolkit.Umbraco 2.2.0' from Umbraco
Successfully uninstalled 'SeoToolkit.Umbraco.Common 2.2.0' from Umbraco
Successfully uninstalled 'SeoToolkit.Umbraco.MetaFields 2.2.0' from Umbraco
Successfully uninstalled 'SeoToolkit.Umbraco.Redirects 2.2.0' from Umbraco
Successfully uninstalled 'SeoToolkit.Umbraco.RobotsTxt 2.2.0' from Umbraco
Successfully uninstalled 'SeoToolkit.Umbraco.ScriptManager 2.2.0' from Umbraco
Successfully uninstalled 'SeoToolkit.Umbraco.SiteAudit 1.0.0' from Umbraco
Successfully uninstalled 'SeoToolkit.Umbraco.SiteAudit.Core 1.0.0' from Umbraco
Successfully uninstalled 'SeoToolkit.Umbraco.Sitemap 2.2.0' from Umbraco
Successfully installed 'SeoToolkit.Umbraco 3.0.0' to Umbraco
Executing nuget actions took 435 ms
Time Elapsed: 00:00:01.3280812
========== Finished ==========
```
(I'm assuming it didn't uninstall `SeoToolkit.Umbraco.Core` 2.2.0 since I have another project still referencing that package)

# TLDR

1. Does this package need to target .NET 7, or can it drop back to 6 (or possibly even 5 for Umbraco 9 users)?
2. Is there some sort of Nuget misconfiguration that is allowing me to upgrade `SeoToolkit.Umbraco` to 3.0.0, even though I'm not meeting the .NET 7 requirement?

**Comment by patrickdemooij9** [comment]
> Does this package need to target .NET 7, or can it drop back to 6 (or possibly even 5 for Umbraco 9 users)?

It does need to target .NET 7 as Umbraco 11 only targets .NET 7. I could take a look if it is possible for the package to be .NET 6, but I am not sure what this would do with the dependency on Umbraco. 

> Is there some sort of Nuget misconfiguration that is allowing me to upgrade SeoToolkit.Umbraco to 3.0.0, even though I'm not meeting the .NET 7 requirement?

Yeah, I think something is going wrong there. The 3.x versions should only be for Umbraco 11 while the 2.x versions should be for Umbraco 10. So if you are using Umbraco 10, then you can stay on the 2.x versions (which will still be updated as long as Umbraco 10 is LTS)

**Comment by ianleeder** [comment]
Yes, I forgot about the dependency in the other direction (SEO on Umbraco).

I wasn't sure if you intended to split SEO Toolkit versions for different Umbraco versions, but I didn't see any documentation to indicate that.  Some other Umbraco packages have synchronised with Umbraco versioning (eg uSync v10 works with Umbraco v10), otherwise I guess you need a table (v2 works with v10, v3 with v11).

I suspected that v3 might only be for v11, but I thought I'd try and update before logging the OpenGraph issue #142

**Comment by ianleeder** [comment]
My apologies @patrickdemooij9 , I just read the `readme` more closely and saw you had included this version information explicitly:
> If you are installing SeoToolkit for Umbraco 9, use the 1.x versions. If you are installing for Umbraco 10, use the 2.x versions. If you are installing for Umbraco 11, use the 3.x versions

**Comment by patrickdemooij9** [comment]
This should now correctly work. I've added a requirement of Umbraco 10 for the 2.x.x versions and a requirement of Umbraco 11 for the 3.x.x versions. It'll go wrong with the 3.0.x versions as I can't add the requirement to them anymore. I'm also going to see if I am able to multi-target the package, but that'll probably be a lot more work

**Comment by ianleeder** [comment]
I think this may not be working correctly yet.  I just installed v3.1.0 into my project with Umbraco 10 (and .NET 6):  
![image](https://user-images.githubusercontent.com/8241300/216849285-c4991904-e688-4116-9f50-5e8573a7ecf6.png)  
(Ignore `.Core`, that's in a different project.)

**Comment by patrickdemooij9** [comment]
That is so confusing. It clearly shows that Umbraco 11 is required, so I am not sure why this would still happen.
<img width="299" alt="image" src="https://user-images.githubusercontent.com/11466511/216944770-a1c87190-1a2e-45d6-9746-12bf55a04711.png">
I guess it'll need some more investigating then

**Comment by patrickdemooij9** [comment]
I am going to be closing this issue as I don't think it's that relevant anymore
